Clarify `ocamldebug` availability

MisterDA opened the pull request:
Quoting a hacker when asked if there was a platform where ocamldebug was unsupported:

> Hah, no - there used to be a requirement for having sockets support 
> The Win32 comment I think dates from before Sylvain implemented `Unix.select` fully - and I think at the same time did the necessary work to disable the fork-y parts of ocamldebug

No change entry needed.

tmcgilchrist commented:
Will this be supported on all the Windows ports? Have you confirmed that it will work with https://github.com/hackwaly/ocamlearlybird? I don't have the hardware available to test it myself.

ghost commented:
Or maybe write "...is available only on systems providing BSD sockets" (i.e. drop "Unix")

nojb reviewed (commented):
The bytecode debugger has been supported on MSVC Windows for more than 15 years, modulo the "replay" facility (going back on time) which is based on `fork`. (There is a note to this effect in `README.win32.adoc`.)

nojb commented on manual/src/cmds/debugger.etex:
```diff
@@ -4,14 +4,6 @@
-
-\begin{windows} The debugger is available under the Cygwin port of
-OCaml, but not under the native Win32 ports.
-\end{windows}
```
Maybe mention here that the "replay" facility is not supported on Windows.

tmcgilchrist replied:
What is the reason for "replay" not being available on Windows?

nojb replied:
As [mentioned](https://github.com/ocaml/ocaml/pull/13087#pullrequestreview-1990920068), this is due to its dependence on `fork`.

nojb commented:
> The bytecode debugger has been supported on MSVC Windows for more than 15 years, modulo the "replay" facility (going back on time) which is based on `fork`. (There is a note to this effect in `README.win32.adoc`.)

(Support was added in 776ae225a0cc5fa44b9279f81d45e9fd3dfa3cca)

MisterDA committed:
Clarify ocamldebug availability

MisterDA commented:
Changed to:
> The debugger is available on systems that provide BSD sockets,
including Windows. On Windows, the "replay" functions are not enabled.

MisterDA committed:
Fix typos in win32 readme

- Fix asciidoc rendering of "C++" text. It gets misinterpreted in some
  contexts, use an asciidoctor intrinsic instead.
- Update links.

eutro commented:
For what it's worth, `README.win32.adoc` does claim that the replay debugger works on the Cygwin version, which does sound correct since it [emulates `fork`](https://cygwin.com/faq.html#faq.api.fork), though it may be incredibly slow since it doesn't use COW.

MisterDA commented:
> For what it's worth, `README.win32.adoc` does claim that the replay debugger works on the Cygwin version, which does sound correct since it [emulates `fork`](https://cygwin.com/faq.html#faq.api.fork), though it may be incredibly slow since it doesn't use COW.

Well noted, now the text mentions only the native Windows ports.
I've piggy-backed a couple of typo fixes.

nojb reviewed (approved):
LGTM

shindere commented:
Sorry for the late response!

Antonin Décimo (2024/04/10 07:37 -0700):
> Changed to:
> > The debugger is available on systems that provide BSD sockets,
> including Windows. On Windows, the "replay" functions are not enabled.

re:replay, shoulnd't we say `available` rather than ` enabled`, which,
to me, suggests that the feature could be enabled, whereas this is not
the case as I understand it?
